includes/options/class-amp-options-manager.php

```diff
@@ -198,6 +200,12 @@ public static function validate_options( $new_options ) {
 		// Store the current version with the options so we know the format.
 		$options['version'] = AMP__VERSION;
 
+		// Handle the caching option.
+		$options['enable_response_caching'] = ! empty( $new_options['enable_response_caching'] );
```

Comment:
The condition here should probably be `wp_using_ext_object_cache() && ! empty( $new_options['enable_response_caching'] )`